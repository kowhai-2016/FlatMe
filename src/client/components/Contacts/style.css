.Contacts #contacts-title {
  font-size: 2.5em;
  text-align: center;
  padding-top: 10px;
}
