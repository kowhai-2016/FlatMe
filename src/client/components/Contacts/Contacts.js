import React, { PropTypes } from 'react'

import './style.css'

const flatmates = props => {
  if (!props.flat) {
    return []
  }

  return props.flat.flatmates.map((flatmate, index) => {
    return (
      <div key={index} className='row'>
        <div className='col-sm-8'>
          <div className='card'>
            <div className='card-block'>
              <h3 className='card-title'>{flatmate.firstName} {flatmate.lastName}</h3>
              <li className='list-group-item'><img src='/images/envelope.svg' width='25px' /> {flatmate.email}</li>
              <li className='list-group-item'><img src='/images/call-volume.svg' width='25px' /> {flatmate.phoneNumber}</li>
              <a href='#' className='btn btn-primary'>Send messages</a>
            </div>
          </div>
        </div>
      </div>
    )
  })
}

const Contacts = props => {
  return (
    <div className='Contacts'>
      <h1 id='contacts-title'>Flatmates</h1>
      <ul>
        {flatmates(props)}
      </ul>
    </div>
  )
}

Contacts.propTypes = {
  flat: PropTypes.object
}

export default Contacts
