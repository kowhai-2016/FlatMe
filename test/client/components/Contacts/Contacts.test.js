/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Contacts from '../../../../src/client/components/Contacts'

test('Contacts renders correctly', () => {
  const tree = renderer.create(
    <Contacts />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
