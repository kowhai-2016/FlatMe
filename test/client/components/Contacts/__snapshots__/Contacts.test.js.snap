exports[`test Contacts renders correctly 1`] = `
<div
  className="Contacts">
  <h1
    id="contacts-title">
    Flatmates
  </h1>
  <ul />
</div>
`;
